### ISSUE TITLE
Implementing mask matrix for UniLM pre-training

### ISSUE THREAD

**Issue by alibi123** [microsoft/unilm#240]
**Describe**
Model I am using (UniLM, MiniLM, LayoutLM ...): UniLM

I want to pre-train UniLM on a custom dataset, but I'm struggling with how to pass the mask matrix for attention. I want to use HuggingFace's BertForPretraining and don't know how to make the mask matrix in the model.
By mask matrix I mean M(ij) from the picture.

<img width="553" alt="Screen Shot 2020-09-18 at 11 23 42 AM" src="https://user-images.githubusercontent.com/51895254/93548069-6fdc9780-f9a1-11ea-8821-39e2eabd474d.png">

**Comment by donglixp** [comment]
https://github.com/microsoft/unilm/blob/master/unilm-v1/src/pytorch_pretrained_bert/modeling.py#L1213
